Parser panic: Negative column value causes panic in error formatting for early returns

## Description

When a function has an early `return` statement directly after the function opening (before the first substantive statement), the parser panics with a Go runtime error instead of properly reporting an unreachable code warning.

## Reproduction

In `examples/tests.ez`, the `test_variables()` function demonstrates the issue:

```ez
do test_variables() -> int{
    using std
    
    return 0
    println("Testing variables...")  // <-- This triggers the panic
    // ... rest of function
}
```

## Error Output

```
panic: strings: negative Repeat count

goroutine 1 [running]:
strings.Repeat({0x100d11928?, 0xc?}, 0x14000159ac0?)
	/usr/local/go/src/strings/strings.go:569 +0x588
github.com/marshallburns/ez/pkg/errors.FormatError(0x1400012e320)
	/Users/marshallburns/code/Project-Zen/pkg/errors/format.go:56 +0x720
```

## Root Cause

The issue has two parts:

1. **Parser issue (pkg/parser/parser.go:808-817)**: When creating the W2001 unreachable code warning, the parser uses `p.currentToken.Column` which is **0** for statements immediately following a return.

2. **Formatter issue (pkg/errors/format.go:56)**: The error formatter calculates:
   ```go
   pointerPadding := strings.Repeat(" ", err.Column-1)
   ```
   When `err.Column` is 0, this becomes `strings.Repeat(" ", -1)`, causing the panic.

## Why Moving the Return Fixes It

When the `return` statement is moved to after the first `println()`, the unreachable code is no longer the first statement, so `p.currentToken.Column` has a valid positive value, and the error formats correctly.

## Expected Behavior

The parser should either:
1. Ensure `p.currentToken.Column` is always >= 1, OR
2. The error formatter should handle edge cases where `Column` is 0 or invalid

The warning should display as:

```
warning[W2001]: unreachable code after return/break/continue
  --> examples/tests.ez:93:X
   |
93 |     println("Testing variables...")
   |     ^ Unreachable code
   |
   = help: remove this code or restructure your control flow
```

## Suggested Fixes

### Option 1: Guard in error formatter (safer, handles all edge cases)
In `pkg/errors/format.go:56`, add validation:
```go
column := err.Column
if column < 1 {
    column = 1
}
pointerPadding := strings.Repeat(" ", column-1)
```

### Option 2: Fix at parser level
In `pkg/parser/parser.go:808-817`, ensure the token has a valid column before creating the warning, or explicitly set one.

## Impact

- **Severity**: High - Causes complete compiler crash with cryptic Go panic
- **Frequency**: Occurs whenever there's an early return before the first statement
- **User Impact**: Users get a confusing panic instead of a helpful warning about unreachable code

--- a/pkg/errors/format.go
+++ b/pkg/errors/format.go
@@ -53,8 +53,13 @@ func FormatError(err *EZError) string {
 			BoldBlue, lineNumStr, Reset, err.SourceLine))
 
 		// Line 5: pointer/underline
-		pointerPadding := strings.Repeat(" ", err.Column-1)
-		pointerLen := err.EndColumn - err.Column
+		// Guard against negative or zero column values to prevent panic
+		column := err.Column
+		if column < 1 {
+			column = 1
+		}
+		pointerPadding := strings.Repeat(" ", column-1)
+		pointerLen := err.EndColumn - column
 		if pointerLen < 1 {
 			pointerLen = 1
 		}

